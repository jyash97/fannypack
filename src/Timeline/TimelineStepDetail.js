// @flow
import React, { type Node } from 'react';
import Box from 'reakit/Box';

type Props = {
  children: Node
};

const TimelineStepDetail = ({ children }: Props) => {
  return <Box>{children}</Box>;
};

export default TimelineStepDetail;
