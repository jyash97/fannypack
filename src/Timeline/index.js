export { default as Timeline } from './Timeline';
export { default } from './Timeline';
