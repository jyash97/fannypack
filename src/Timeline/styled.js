import styled from 'reakit/styled';
import { palette } from 'styled-tools';
import Box from 'reakit/Box';

export const Row = styled(Box)`
  display: flex;
  position: relative;

  &:last-child {
    margin-bottom: 0;
  }

  &::before {
    content: '';
    border: 1px solid ${props => props.lineColor || palette('textTint')};
    position: absolute;
    height: 100%;
    left: 0.25rem;
    margin-top: 1.25rem;
  }

  &:last-child::before {
    border: 0;
  }
`;

export const StepBullet = styled.span`
  background-color: ${props => props.color || palette('textTint')};
  border-radius: 50%;
  display: inline-flex;
  height: 0.625rem;
  width: 0.625rem;
  margin-top: 1rem;
  margin-right: 0.5rem;
  
  ${theme('fannypack.Step.bullet.base')}
`;

export const StepLabel = styled(Box)`
  margin-top: 0.6rem;
  line-height: 1.5rem;
  
  ${theme('fannypack.Step.Label.base')}
`;
