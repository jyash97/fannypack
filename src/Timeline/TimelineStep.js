// @flow
import React, { type Node } from 'react';
import Box from 'reakit/Box';

import { StepLabel } from './styled';

type Props = {
  label: string,
  color?: string,
  children: Node
};

const TimelineStep = ({ label, color, children }: Props) => {
  return (
    <Box>
      <StepLabel color={color}>{label}</StepLabel>
      {children}
    </Box>
  );
};

TimelineStep.defaultProps = {
  color: undefined
};

export default TimelineStep;
