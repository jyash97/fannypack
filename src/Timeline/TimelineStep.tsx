import * as React from 'react';
import * as PropTypes from 'prop-types';

import { TimelineStep as _TimelineStep, TimelineRow, TimelineBullet } from './styled';

export type LocalTimelineStepProps = {
  a11yDescriptionId?: string;
  a11yTitleId?: string;
  children: React.ReactNode;
  lineColor?: string;
  progressColor?: string;
};
export type LocalTimelineRowProps = {
  lineColor?: string;
};

export type TimelineStepProps = LocalTimelineStepProps;
export type TimelineRowProps = LocalTimelineRowProps;

const TimelineStep: React.FunctionComponent<LocalTimelineStepProps> = ({
  a11yDescriptionId,
  a11yTitleId,
  children,
  lineColor,
  progressColor
}) => {
  return (
    <TimelineRow lineColor={lineColor} aria-labelledby={a11yTitleId} aria-describedby={a11yDescriptionId}>
      <TimelineBullet backgroundColor={progressColor} />
      <_TimelineStep aria-labelledby={a11yTitleId} aria-describedby={a11yDescriptionId}>
        {children}
      </_TimelineStep>
    </TimelineRow>
  );
};

TimelineStep.propTypes = {
  a11yDescriptionId: PropTypes.string,
  a11yTitleId: PropTypes.string,
  children: PropTypes.node.isRequired,
  lineColor: PropTypes.string,
  progressColor: PropTypes.string
};

TimelineStep.defaultProps = {
  a11yDescriptionId: undefined,
  a11yTitleId: undefined,
  lineColor: undefined,
  progressColor: 'textTint'
};

// @ts-ignore
const C: React.FunctionComponent<TimelineStepProps> = TimelineStep;
export default C;
