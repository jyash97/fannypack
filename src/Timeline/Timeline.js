// @flow
import React, { Component, type Node } from 'react';
import { Box } from '../primitives';

import TimelineStep from './TimelineStep';
import TimelineStepDetail from './TimelineStepDetail';
import { Row, StepBullet } from './styled';

type TimelineProps = {
  children: Node
};

class Timeline extends Component<TimelineProps> {
  static Step = TimelineStep;
  static StepDetail = TimelineStepDetail;

  render() {
    const { children } = this.props;
    return (
      <Box>
        {React.Children.map(children, (child, index) => {
          const color = child.props.color;
          return (
            <Row lineColor={color}>
              <StepBullet color={color} />
              {child}
            </Row>
          );
        })}
      </Box>
    );
  }
}

export default Timeline;
