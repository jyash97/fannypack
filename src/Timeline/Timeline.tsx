import * as React from 'react';
import * as PropTypes from 'prop-types';

import TimelineStep, { TimelineStepProps } from './TimelineStep';
import TimelineStepDetail, { TimelineStepDetailProps } from './TimelineStepDetail';
import TimelineStepHeading, { TimelineStepHeadingProps } from './TimelineStepHeading';
import { Timeline as _Timeline } from './styled';

export type LocalTimelineProps = {
  activeStep?: string;
  children?: React.ReactNode;
  lineColor?: string;
  progressColor?: string;
};
export type TimelineProps = LocalTimelineProps;
export type TimelineComponents = {
  Step: React.FunctionComponent<TimelineStepProps>;
  StepHeading: React.FunctionComponent<TimelineStepHeadingProps>;
  StepDetail: React.FunctionComponent<TimelineStepDetailProps>;
};

export const Timeline: React.FunctionComponent<LocalTimelineProps> & TimelineComponents = ({
  activeStep,
  children,
  lineColor,
  progressColor
}) => {
  const activeIndexStep = children.findIndex(child => child.props.a11yTitleId === activeStep);
  return (
    <_Timeline>
      {React.Children.map(children, (child, index) => {
        const barColor = lineColor ? lineColor : child.props.lineColor;
        const dotColor = progressColor ? progressColor : child.props.progressColor;
        return index <= activeIndexStep
          ? React.cloneElement(child, {
              lineColor: index < activeIndexStep ? barColor : undefined,
              progressColor: dotColor || undefined
            })
          : child;
      })}
    </_Timeline>
  );
};

Timeline.Step = TimelineStep;
Timeline.StepHeading = TimelineStepHeading;
Timeline.StepDetail = TimelineStepDetail;

Timeline.propTypes = {
  children: PropTypes.node.isRequired,
  activeStep: PropTypes.string,
  progressColor: PropTypes.string,
  lineColor: PropTypes.string
};
Timeline.defaultProps = {
  activeStep: undefined,
  progressColor: undefined,
  lineColor: undefined
};

// @ts-ignore
const C: React.FunctionComponent<TimelineProps> & TimelineComponents = Timeline;
export default C;
