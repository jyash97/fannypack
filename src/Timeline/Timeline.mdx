---
name: Timeline
route: /components/timeline
menu: Components
---

import { Playground, PropsTable } from 'docz';
import Timeline from './index';


# Timeline

## Import

`import { Timeline } from 'fannypack'`

## Basic Usage

The `<Timeline>` component is made up of a compound of components such as: `<Timeline.Step>`,  `<Timeline.StepHeading>` and `<Timeline.StepDetail>`

<Playground>
  <Timeline activeStep="StepHeading2" progressColor="green" lineColor="green">
    <Timeline.Step a11yTitleId="StepHeading1" a11yDescriptionId="StepDetail1">
      <Timeline.StepHeading a11yTitleId="StepHeading1">Step one Heading</Timeline.StepHeading>
      <Timeline.StepDetail a11yDescriptionId="StepDetail1">Step one details goes here</Timeline.StepDetail>
    </Timeline.Step>
    <Timeline.Step a11yTitleId="StepHeading2" a11yDescriptionId="StepDetail2">
      <Timeline.StepHeading a11yTitleId="StepHeading2">Step two Heading</Timeline.StepHeading>
      <Timeline.StepDetail a11yDescriptionId="StepDetail2">Step two details goes here</Timeline.StepDetail>
    </Timeline.Step>
    <Timeline.Step a11yTitleId="StepHeading3" a11yDescriptionId="StepDetail3">
      <Timeline.StepHeading a11yTitleId="StepHeading3">Step three Heading</Timeline.StepHeading>
      <Timeline.StepDetail a11yDescriptionId="StepDetail3">Step three details goes here</Timeline.StepDetail>
    </Timeline.Step>
  </Timeline>
</Playground>

## Props

### `<Timeline>`
<PropsTable of={Timeline} />

### `<Timeline.Step>`
<PropsTable of={Timeline.Step} />

### `<Timeline.StepHeading>`
<PropsTable of={Timeline.StepHeading} />

### `<Timeline.StepDetail>`
<PropsTable of={Timeline.StepDetail} />


## Theming

### Schema

```jsx
{
  base: string | Object,
  Bullet: {
    base: string | Object
  },
  StepHeading: {
    base: string | Object
  },
  StepDetail: {
    base: string | Object
  }
}
```
