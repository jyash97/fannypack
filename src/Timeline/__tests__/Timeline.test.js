import React from 'react';
import render from '../../_utils/tests/render';
import Timeline from '../Timeline';

it('renders Timeline correctly', () => {
  const { container } = render(
    <Timeline>
      <Timeline.Step a11yTitleId="StepHeading1" a11yDescriptionId="StepDetail1">
        <Timeline.StepHeading a11yTitleId="StepHeading1">Step one Heading</Timeline.StepHeading>
        <Timeline.StepDetail a11yDescriptionId="StepDetail1">Step one details goes here</Timeline.StepDetail>
      </Timeline.Step>
      <Timeline.Step a11yTitleId="StepHeading2" a11yDescriptionId="StepDetail2">
        <Timeline.StepHeading a11yTitleId="StepHeading2">Step two Heading</Timeline.StepHeading>
        <Timeline.StepDetail a11yDescriptionId="StepDetail2">Step two details goes here</Timeline.StepDetail>
      </Timeline.Step>
      <Timeline.Step a11yTitleId="StepHeading3" a11yDescriptionId="StepDetail3">
        <Timeline.StepHeading a11yTitleId="StepHeading3">Step three Heading</Timeline.StepHeading>
        <Timeline.StepDetail a11yDescriptionId="StepDetail3">Step three details goes here</Timeline.StepDetail>
      </Timeline.Step>
    </Timeline>
  );
  expect(container).toMatchSnapshot();
});
