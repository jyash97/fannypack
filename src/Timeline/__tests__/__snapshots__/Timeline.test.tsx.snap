// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline renders correctly 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0:focus:not(:focus-visible) {
  outline: none;
}

.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
  background-color: #d9dee2 !important;
}

.c3:focus:not(:focus-visible) {
  outline: none;
}

.c6 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
  font-weight: 600 !important;
}

.c6:focus:not(:focus-visible) {
  outline: none;
}

.c5:not(:last-child) {
  margin: 0 0 1rem;
}

.c5 .c8 {
  top: 0.15em;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.c1:last-child {
  margin-bottom: 0;
}

.c1::before {
  content: '';
  border: 1px solid #d9dee2;
  position: absolute;
  height: 100%;
  left: 0.25rem;
  margin-top: 1.25rem;
}

.c1:last-child::before {
  border: 0;
}

.c2 {
  border-radius: 50%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 0.625rem;
  width: 0.625rem;
  margin-top: 1rem;
  margin-right: 0.5rem;
  z-index: 1;
}

.c4 {
  margin-top: 0.6rem;
  margin-bottom: 0.6rem;
  line-height: 1.5rem;
  color: #435a6f;
}

.c4:not(:last-child) {
  margin-top: 0.6rem;
  margin-bottom: 0.6rem;
}

.c7 {
  color: #435a6f;
}

<div
  class="c0"
>
  <div
    aria-describedby="StepDetail1"
    aria-labelledby="StepHeading1"
    class="c1 c0"
  >
    <div
      class="c2 c3"
    />
    <div
      aria-describedby="StepDetail1"
      aria-labelledby="StepHeading1"
      class="c0"
    >
      <p
        class="c4 c5 c6"
        font-weight="semibold"
        id="StepHeading1"
        style="font-weight: semibold;"
      >
        Step one Heading
      </p>
      <p
        class="c7 c5 c0"
        id="StepDetail1"
      >
        Step one details goes here
      </p>
    </div>
  </div>
  <div
    aria-describedby="StepDetail2"
    aria-labelledby="StepHeading2"
    class="c1 c0"
  >
    <div
      class="c2 c3"
    />
    <div
      aria-describedby="StepDetail2"
      aria-labelledby="StepHeading2"
      class="c0"
    >
      <p
        class="c4 c5 c6"
        font-weight="semibold"
        id="StepHeading2"
        style="font-weight: semibold;"
      >
        Step two Heading
      </p>
      <p
        class="c7 c5 c0"
        id="StepDetail2"
      >
        Step two details goes here
      </p>
    </div>
  </div>
  <div
    aria-describedby="StepDetail3"
    aria-labelledby="StepHeading3"
    class="c1 c0"
  >
    <div
      class="c2 c3"
    />
    <div
      aria-describedby="StepDetail3"
      aria-labelledby="StepHeading3"
      class="c0"
    >
      <p
        class="c4 c5 c6"
        font-weight="semibold"
        id="StepHeading3"
        style="font-weight: semibold;"
      >
        Step three Heading
      </p>
      <p
        class="c7 c5 c0"
        id="StepDetail3"
      >
        Step three details goes here
      </p>
    </div>
  </div>
</div>
`;

exports[`Timeline with activeStep renders correctly 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0:focus:not(:focus-visible) {
  outline: none;
}

.c8 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
  background-color: #d9dee2 !important;
}

.c8:focus:not(:focus-visible) {
  outline: none;
}

.c5 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
  font-weight: 600 !important;
}

.c5:focus:not(:focus-visible) {
  outline: none;
}

.c4:not(:last-child) {
  margin: 0 0 1rem;
}

.c4 .c9 {
  top: 0.15em;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.c7:last-child {
  margin-bottom: 0;
}

.c7::before {
  content: '';
  border: 1px solid #d9dee2;
  position: absolute;
  height: 100%;
  left: 0.25rem;
  margin-top: 1.25rem;
}

.c7:last-child::before {
  border: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.c1:last-child {
  margin-bottom: 0;
}

.c1::before {
  content: '';
  border: 1px solid purple;
  position: absolute;
  height: 100%;
  left: 0.25rem;
  margin-top: 1.25rem;
}

.c1:last-child::before {
  border: 0;
}

.c2 {
  border-radius: 50%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 0.625rem;
  width: 0.625rem;
  margin-top: 1rem;
  margin-right: 0.5rem;
  z-index: 1;
}

.c3 {
  margin-top: 0.6rem;
  margin-bottom: 0.6rem;
  line-height: 1.5rem;
  color: #435a6f;
}

.c3:not(:last-child) {
  margin-top: 0.6rem;
  margin-bottom: 0.6rem;
}

.c6 {
  color: #435a6f;
}

<div
  class="c0"
>
  <div
    aria-describedby="StepDetail1"
    aria-labelledby="StepHeading1"
    class="c1 c0"
  >
    <div
      class="c2 c0"
      style="background-color: purple;"
    />
    <div
      aria-describedby="StepDetail1"
      aria-labelledby="StepHeading1"
      class="c0"
    >
      <p
        class="c3 c4 c5"
        font-weight="semibold"
        id="StepHeading1"
        style="font-weight: semibold;"
      >
        Step one Heading
      </p>
      <p
        class="c6 c4 c0"
        id="StepDetail1"
      >
        Step one details goes here
      </p>
    </div>
  </div>
  <div
    aria-describedby="StepDetail2"
    aria-labelledby="StepHeading2"
    class="c7 c0"
  >
    <div
      class="c2 c0"
      style="background-color: purple;"
    />
    <div
      aria-describedby="StepDetail2"
      aria-labelledby="StepHeading2"
      class="c0"
    >
      <p
        class="c3 c4 c5"
        font-weight="semibold"
        id="StepHeading2"
        style="font-weight: semibold;"
      >
        Step two Heading
      </p>
      <p
        class="c6 c4 c0"
        id="StepDetail2"
      >
        Step two details goes here
      </p>
    </div>
  </div>
  <div
    aria-describedby="StepDetail3"
    aria-labelledby="StepHeading3"
    class="c7 c0"
  >
    <div
      class="c2 c8"
    />
    <div
      aria-describedby="StepDetail3"
      aria-labelledby="StepHeading3"
      class="c0"
    >
      <p
        class="c3 c4 c5"
        font-weight="semibold"
        id="StepHeading3"
        style="font-weight: semibold;"
      >
        Step three Heading
      </p>
      <p
        class="c6 c4 c0"
        id="StepDetail3"
      >
        Step three details goes here
      </p>
    </div>
  </div>
</div>
`;

exports[`Timeline with custom color in step renders correctly 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0:focus:not(:focus-visible) {
  outline: none;
}

.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
  background-color: #d9dee2 !important;
}

.c3:focus:not(:focus-visible) {
  outline: none;
}

.c6 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
  font-weight: 600 !important;
}

.c6:focus:not(:focus-visible) {
  outline: none;
}

.c5:not(:last-child) {
  margin: 0 0 1rem;
}

.c5 .c9 {
  top: 0.15em;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.c1:last-child {
  margin-bottom: 0;
}

.c1::before {
  content: '';
  border: 1px solid #d9dee2;
  position: absolute;
  height: 100%;
  left: 0.25rem;
  margin-top: 1.25rem;
}

.c1:last-child::before {
  border: 0;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.c8:last-child {
  margin-bottom: 0;
}

.c8::before {
  content: '';
  border: 1px solid green;
  position: absolute;
  height: 100%;
  left: 0.25rem;
  margin-top: 1.25rem;
}

.c8:last-child::before {
  border: 0;
}

.c2 {
  border-radius: 50%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 0.625rem;
  width: 0.625rem;
  margin-top: 1rem;
  margin-right: 0.5rem;
  z-index: 1;
}

.c4 {
  margin-top: 0.6rem;
  margin-bottom: 0.6rem;
  line-height: 1.5rem;
  color: #435a6f;
}

.c4:not(:last-child) {
  margin-top: 0.6rem;
  margin-bottom: 0.6rem;
}

.c7 {
  color: #435a6f;
}

<div
  class="c0"
>
  <div
    aria-describedby="StepDetail1"
    aria-labelledby="StepHeading1"
    class="c1 c0"
  >
    <div
      class="c2 c3"
    />
    <div
      aria-describedby="StepDetail1"
      aria-labelledby="StepHeading1"
      class="c0"
    >
      <p
        class="c4 c5 c6"
        font-weight="semibold"
        id="StepHeading1"
        style="font-weight: semibold;"
      >
        Step one Heading
      </p>
      <p
        class="c7 c5 c0"
        id="StepDetail1"
      >
        Step one details goes here
      </p>
    </div>
  </div>
  <div
    aria-describedby="StepDetail2"
    aria-labelledby="StepHeading2"
    class="c8 c0"
  >
    <div
      class="c2 c0"
      style="background-color: green;"
    />
    <div
      aria-describedby="StepDetail2"
      aria-labelledby="StepHeading2"
      class="c0"
    >
      <p
        class="c4 c5 c6"
        font-weight="semibold"
        id="StepHeading2"
        style="font-weight: semibold;"
      >
        Step two Heading
      </p>
      <p
        class="c7 c5 c0"
        id="StepDetail2"
      >
        Step two details goes here
      </p>
    </div>
  </div>
  <div
    aria-describedby="StepDetail3"
    aria-labelledby="StepHeading3"
    class="c1 c0"
  >
    <div
      class="c2 c3"
    />
    <div
      aria-describedby="StepDetail3"
      aria-labelledby="StepHeading3"
      class="c0"
    >
      <p
        class="c4 c5 c6"
        font-weight="semibold"
        id="StepHeading3"
        style="font-weight: semibold;"
      >
        Step three Heading
      </p>
      <p
        class="c7 c5 c0"
        id="StepDetail3"
      >
        Step three details goes here
      </p>
    </div>
  </div>
</div>
`;
